### DISCUSSION TITLE
Can the Client be secured with TPM 2.0

### DISCUSSION THREAD

**Discussion by bmoe3rd** [strongswan/strongswan#225] [Q&A]
Hi, 

this is probably a silly question, but  i am currently evaluating VPN Solutions
and need a definitive answer, if strongSwan is able to use TPMs for the Server only
or for Clients too?

Best, bmoe3rd

**Comment by strongX509** [comment]
You can use a TPM 2.0 for storing your private keys both for servers and clients. If have been doing this on my systems for a couple of years.

Best regards

Andreas

**Comment by bmoe3rd** [comment]
Thank you for your helpful reply,

Best regards :)
